src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -76,6 +76,16 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
                 }
 
                 var diagnosticSpan = diagnostic.Range.AsTextSpan(context.SourceText);
+
+                // Based on how we compute `Range.AsTextSpan` it's possible to have a span
+                // which goes beyond the end of the source text. Something likely changed
+                // between the capturing of the diagnostic (by the platform) and the retrieval of the 
+                // document snapshot / source text. In such a case, we skip processing of the diagnostic.
+                if (diagnosticSpan.End > context.SourceText.Length)
```

Comment:
Should this be at a higher layer so we don't try computing any light bulbs that have this apply?

Reply:
This is the only provider for which we have to extract out the underlying string value covered by the diagnostic. We use this to determine heuristically whether a FQN/add using is appropriate. This should all go away once we have some kind of identifier from roslyn of code action kind.